monai/inferers/splitter.py

```diff
@@ -58,142 +91,370 @@ def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
 
-        if any(pad_size):
+        # Padding
+        if self.pad_mode and any(pad_size):
             # pad the inputs
-            inputs = torch.nn.functional.pad(
-                inputs, pad_size[::-1], look_up_option(self.pad_mode, PytorchPadMode).value, **self.pad_kwargs
-            )
+            inputs = torch.nn.functional.pad(inputs, pad_size[::-1], mode=self.pad_mode, value=self.pad_value)
             # update spatial shape
-            spatial_shape = inputs.shape[n_non_spatial_dims:]
+            spatial_shape = inputs.shape[2:]
+            # correct the offset with respect to the padded image
+            if is_start_padded:
+                offset = tuple(off + p for off, p in zip(offset, pad_size[1::2]))
+
+        # Splitting
+        for location in iter_patch_position(spatial_shape, patch_size, offset, overlap, False):
+            patch = self._get_patch(inputs, location, patch_size)  # type: ignore
+            patch = ToTensor(device=self.device)(patch)  # type: ignore
+            # correct the location with respect to original inputs (remove starting pads)
+            if is_start_padded:
+                location = tuple(loc - p for loc, p in zip(location, pad_size[1::2]))
+            # filter patch and yield
+            if self.filter_fn is None:
+                yield patch, location
+            elif self.filter_fn(patch, location):
+                yield patch, location
+
+
+class WSISlidingWindowSplitter(SlidingWindowSplitter):
+    """Split the whole slide image input into patches with sliding window strategy and a possible overlap.
+    This extract patches from file without loading the entire slide into memory.
+    It also allows to offset the starting position and filter the patches.
+
+    Args:
+        patch_size : the size of the patches to be generated.
+        offset: the amount of offset for the patches with respect to the original input.  Defaults to 0.
+        overlap: the amount of overlap between patches in each dimension. It can be either a float in
+            the range of [0.0, 1.0) that defines relative overlap to the patch size, or it can be a non-negative int
+            that defines number of pixels for overlap. Defaults to 0.0.
+        filter_fn: a callable to filter patches. It should accepts exactly two parameters (patch, location), and
+            return True for a patch to keep. Defaults to no filtering.
+        pad_mode: define the mode for padding. Either "constant" or None. Default to None.
+            Padding is only supported with "OpenSlide" or "cuCIM" backend, and the filling value is 256.
+        device: the device where the patches are generated. Defaults to the device of inputs.
+        reader: the module to be used for loading whole slide imaging. If `reader` is
+
+            - a string, it defines the backend of `monai.data.WSIReader`. Defaults to cuCIM.
```

Comment:
Seems the default reader is not `cuCIM` in the code.
Could you please help list all the supported readers in the doc-string?

Thanks.

Reply:
Add it in L310.